### PR TITLE
#166840924 Fetch Article Ratings 

### PR DESCRIPTION
**Pull request by tolumide-ng**
#### Description
Users can fetch ratings of there articles  via **/api/v1/articles/ratings/:articleId** on the errorSwag application 

#### Type of Change
- [x] New feature (Non-breaking change which adds functionality)
- [x] Bug fix (Non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Integration Testing

#### Pivotal Stories
[#166840924](https://www.pivotaltracker.com/story/show/166840924)

### CODE REVIEW

File: src/controllers/rating.controller.js [hunk 1]
```diff
@@ -60,6 +61,41 @@ class RatingsController {
...
+   * @async
+   * @param {object} req - Request object
+   * @param {object} res - Response object
+   * @return {json} Returns containing all ratings of an article
+   * @static
+   */
+  static async getRatings(req, res) {
+    const { articleId } = req.params;
+
+    const averageRating = await BaseRepository.averageRating(
```
Comment (timi-codes) [inline src/controllers/rating.controller.js]: `averageRating` doesn't describe what this does?

File: src/repository/base.repository.js [hunk 2]
```diff
@@ -182,6 +197,30 @@ class BaseRepository {
...
+   * @static
+   * @param {*} articleId - what to searchBy
+   * @param {*} model - model to search
+   * @param {*} searchBy - column name of what to seach by
+   * @param {*} value - column name of what to calculate average by
+   * @returns {object} - returns an database object
+   * @memberof BaseRepository
+   */
+  static async averageRating(articleId, model, searchBy, value) {
+    const abg = model.findAll({
```
Comment (Onyimatics) [inline src/repository/base.repository.js]: I don't think I understand the name of this variable.

**Review by Onyimatics — CHANGES_REQUESTED** [review]
Good job @tolumide-ng. This looks good to me. However, I left a comment.

File: src/repository/base.repository.js (hunk 2 above)
Comment (tolumide-ng) [reply src/repository/base.repository.js]: Thank you, I would resolve this now

**Review by Onyimatics — APPROVED** [review]

File: src/controllers/rating.controller.js [hunk 3]
```diff
@@ -60,6 +61,41 @@ class RatingsController {
...
+      );
+    }
+
+    const summary = {
+      articleId: response.articleId,
+      averageRating: Number(response.avgRating).toFixed(2),
+      totalNumberOfRatings: Number(response.totalCount).toFixed(2)
+    };
+
+    return responseGenerator.sendSuccess(res, 200, null, { ...summary });
```
Comment (timi-codes) [inline src/controllers/rating.controller.js]: The signature of the response generator takes data as the third argument  - `sendSuccess(res, statusCode, data, message = '', metadata = '')`. the summary should be the third argument

**Review by timi-codes — CHANGES_REQUESTED** [review]
Great Job!, I left some comments for you

File: src/controllers/rating.controller.js (hunk 1 above)
Comment (codexempire) [reply src/controllers/rating.controller.js]: `BaseRepository.averageRating` nice, but please could you rename the method to `getAverage` so as to make it generalistic and reusable for others

File: src/controllers/rating.controller.js [hunk 4]
```diff
@@ -60,6 +61,41 @@ class RatingsController {
...
+      return responseGenerator.sendError(
+        res,
+        404,
+        'This article has not been rated yet'
+      );
+    }
+
+    const summary = {
+      articleId: response.articleId,
+      averageRating: Number(response.avgRating).toFixed(2),
```
Comment (codexempire) [inline src/controllers/rating.controller.js]: Please can the fix `averageRating` and make it `1 decimal place` number

File: src/controllers/rating.controller.js [hunk 5]
```diff
@@ -60,6 +61,41 @@ class RatingsController {
...
+      totalNumberOfRatings: Number(response.totalCount).toFixed(2)
```
Comment (codexempire) [inline src/controllers/rating.controller.js]: Can the `totalNumberOfRatings` be rounded up to a whole number without the decimals.

File: src/controllers/rating.controller.js (hunk 1 above)
Comment (tolumide-ng) [reply src/controllers/rating.controller.js]: Thank you so much, I will implement changes

File: src/test/rating.spec.js [hunk 6]
```diff
@@ -112,3 +117,77 @@ describe('PATCH, /api/v1/articles/:id/ratings', () => {
...
+      ratings: 1
+    });
+
+    const numberOfRatings = await BaseRepository.findRawAndCountAll(db.Rating, {
+      articleId: article.id
+    });
+
+    const res = await server().get(`${ARTICLE_API}/ratings/${article.id}`);
+    expect(res.status).to.equal(200);
+    expect(Number(`${res.body.message.totalNumberOfRatings}`)).to.equal(
```
Comment (codexempire) [inline src/test/rating.spec.js]: I belief this `${res.body.message.totalNumberOfRatings}` could be represented as `res.body.message.totalNumberOfRatings`

**Review by codexempire — CHANGES_REQUESTED** [review]
Nice work 👍 please refer to the comments

**Pull request closed without merging**